samples/angular-todo/src/app/app.component.ts

```diff
@@ -65,7 +65,7 @@ export class AppComponent implements OnInit {
 
     Config.set({
       clientId: environment.WEB_OAUTH_CLIENT,
-      redirectUri: `${window.location.origin}/callback`,
```

Comment:
i'm asking about this as if i check the routing file, the path 'callback' does not exist and therefore throws an error
![image](https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/dfbe1022-ddb5-44fe-860a-125b9263abfe)
![image](https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/fc39cd31-a56e-4700-bc47-d783d062d143)

Reply:
I see what Juan is trying to fix, and I think we should address it. The question is just how we fix it. Currently, both the React app and the Angular app use a redirect URL that actually doesn't point to something that's properly handled. The Angular app may just be more sensitive to this, as it results in an error seen in Juan's screenshot.

Ultimately, I don't feel this redirect URL should result in the Angular app itself, or really anything that uses JS and CSS, being loaded and rendered. Maybe we could point to a static file, rather than a "route"? So, instead of `${window.location.origin}/home` or `${window.location.origin}/callback`, we could use `${window.location.origin}/callback.html`. The server should recognize that as a static resource, rather than a route in the main app, which results in quicker load times and less resource usage.

Something along the lines of a plain old HTML file that just renders `<!DOCTYPE html><html><p>OK</p></html>`. That way, the page loads and the SDK can quickly read the final URL of the resolved request. Apps with embedded login will never actually see the rendered page, so this makes sense to me.

Thoughts?

Reply:
i fixed by adding a new path into the routing file, when '/callback' it loads the HomeComponent

Reply:
@cerebrl yeah, it was pointing at a callback.html in the assets directory but it seems angular router doesn't like to not know about possible routes. so Juan was able to correct it by assigning a router so angular can say it knows about the route. That seems to fix the angular issue we had previously.

So if its fixed, i don't think webpack dev server was rewriting the params, it was angular rejecting the route because it was not aware of it.

Reply:
This is a solution, but I'm not sure it's one that we should use long-term. We don't have to address it in this PR, but here's my reasoning:

The request to `/callback` due to the redirect from `/authorize` just has to resolve when it's a part of Embedded Login. It doesn't have to render anything as everything within the iframe will be hidden when using Embedded Login. Due to this, all the resources the full Angular app is dependent on has to be loaded onto the page in order to call the `onload` event found here: https://github.com/ForgeRock/forgerock-javascript-sdk/blob/develop/packages/javascript-sdk/src/oauth2-client/index.ts#L110. This is required for the token request to be continued.

Regarding the `load` event ([from MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event)):

> The load event is fired when the whole page has loaded, including all dependent resources such as stylesheets, scripts, iframes, and images.

To load up an entire Angular application just to resolve a URL that will never be rendered is ... well, a lot. To keep this app as "real world" as we can, I would suggest we do what we would recommend to our customers, which is have `/callback`, or whatever we call it, refer to a static asset, rather than an full JS application.

Now, if this was Centralized Login, it would be a different story as the redirect URL is used in a full-page redirect, rather than a hidden iframe. So, context matters.

Reply:
Hmm, i could have sworn this was the behavior at one point. I don't disagree, so lets see if we can do that Juan.

Add a basic html file (i feel we've done this before) called callback.html. We will need to serve that file as a static asset. 

I think you will add it to the `assets` key in project.json under serve, so it serves callback.html (i'd keep it at the same level of the index.html).

See if we can serve this file without an angular route specifically.

Reply:
done